### PR TITLE
fix #37656: don't always error on space before '

### LINKED ISSUE #37656: How is this conditional a syntax error?
This behavior seems odd to me:

```julia
if true 'a' else 1 end # syntax error
if true "a" else 1 end # returns "a"
if true 1 else 'a' end # returns 1
```

The backtrace is:

```
ERROR: syntax: space not allowed before "'"
Stacktrace:
 [1] top-level scope at none:1
```

### PR DESCRIPTION
**Pull request by simeonschaub**
This slightly changes the error message for things like `a 'b'` in non-space-sensitive contexts from `syntax: space not allowed before "'"` to `syntax: extra token "'" after end of expression`, which I don't think is really a big loss. It just doesn't make much sense to me to explicitly disallow whitespace before `'` in general. I wonder whether we should do the same for `(`, `[` and `{`, since I don't see any reason we should generally disallow whitespace before them as well. Was there a reason this was done in the first place?
fixes #37656

### CODE REVIEW

File: src/julia-parser.scm [hunk 1]
```diff
@@ -1162,7 +1162,7 @@
   (let loop ((ex ex))
     (let ((t (peek-token s)))
       (if (or (and space-sensitive (ts:space? s)
-                   (memv t '(#\( #\[ #\{ |'| #\" #\`)))
+                   (memv t '(#\( #\[ #\{ #\' #\" #\`)))
```
Comment (JeffBezanson) [inline src/julia-parser.scm:1165]: I'm pretty sure the lexer returns ' as a symbol.

Comment (simeonschaub) [reply src/julia-parser.scm:1165]: Ah, ok, I was a bit confused by that. So the lexer returns `(`, `[`, ... as characters and `'` as a token? At first I thought this was causing the issue, but apparently both work. I can change this back.

Comment (simeonschaub) [reply src/julia-parser.scm:1165]:
```suggestion
                   (memv t '(#\( #\[ #\{ |'| #\" #\`)))
```

**Review by JeffBezanson — APPROVED** [review]

**Comment by JeffBezanson** [comment]
On `(`, see #7232. I think parens are a bit different. `if f 'x' end` is pretty clearly the condition `f` followed by a character literal. But, `if f (x) end` is ambiguous since in some languages people do put spaces between function names and parenthesized arguments. So I think we should continue to disallow that.

**Pull request merged**